### DISCUSSION TITLE
Error 500 on notice subtype 29

### DISCUSSION THREAD

**Discussion by rengit62** [OP-TED/eForms-SDK#454] [Q&A] [answered]
Hi lab
I get an 'ugly' error when I send a notice subtype 29 ( CVS Validated of course ) :
org.springframework.web.client.HttpServerErrorException$InternalServerError: 500 Internal Server Error: "<!doctype html><html lang="en"><head><title>HTTP Status 500 – Internal Server Error</title><style type="text/css">body {font-family:Tahoma,Arial,sans-serif;} h1, h2, h3, b {color:white;background-color:#525D76;} h1 {font-size:22px;} h2 {font-size:16px;} h3 {font-size:14px;} p {font-size:12px;} a {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h1>HTTP Status 500 – Internal Server Error</h1></body></html>"
No error sending ( for example ) notice subtype 14, so I guess there is something wrong in the notice , although it was validated from CVS, but with a 500 error I cannot understand this error
Can you help me ?
Thanks and Regards
Renato

`<?xml version="1.0" encoding="UTF-8"?><ContractAwardNotice xmlns="urn:oasis:names:specification:ubl:schema:xsd:ContractAwardNotice-2" xmlns:cac="urn:oasis:names:specification:ubl:schema:xsd:CommonAggregateComponents-2" xmlns:cbc="urn:oasis:names:specification:ubl:schema:xsd:CommonBasicComponents-2" xmlns:efac="http://data.europa.eu/p27/eforms-ubl-extension-aggregate-components/1" xmlns:efbc="http://data.europa.eu/p27/eforms-ubl-extension-basic-components/1" xmlns:efext="http://data.europa.eu/p27/eforms-ubl-extensions/1" xmlns:ext="urn:oasis:names:specification:ubl:schema:xsd:CommonExtensionComponents-2"><ext:UBLExtensions><ext:UBLExtension><ext:ExtensionContent><efext:EformsExtension><efac:NoticeResult><efac:LotResult><cbc:ID schemeName="result">RES-0001</cbc:ID><cbc:TenderResultCode listName="winner-selection-status">selec-w</cbc:TenderResultCode><efac:LotTender><cbc:ID>TEN-0001</cbc:ID></efac:LotTender><efac:SettledContract><cbc:ID>CON-0001</cbc:ID></efac:SettledContract><efac:TenderLot><cbc:ID>LOT-0001</cbc:ID></efac:TenderLot></efac:LotResult><efac:LotTender><cbc:ID schemeName="tender">TEN-0001</cbc:ID><efac:TenderingParty><cbc:ID>TPA-0001</cbc:ID></efac:TenderingParty><efac:TenderLot><cbc:ID>LOT-0001</cbc:ID></efac:TenderLot></efac:LotTender><efac:SettledContract><cbc:ID schemeName="contract">CON-0001</cbc:ID></efac:SettledContract><efac:TenderingParty><cbc:ID schemeName="tendering-party">TPA-0001</cbc:ID><efac:Tenderer><cbc:ID schemeName="organization">ORG-0002</cbc:ID></efac:Tenderer></efac:TenderingParty></efac:NoticeResult><efac:NoticeSubType><cbc:SubTypeCode listName="notice-subtype">29</cbc:SubTypeCode></efac:NoticeSubType><efac:Organizations><efac:Organization><efac:Company><cac:PartyIdentification><cbc:ID schemeName="organization">ORG-0001</cbc:ID></cac:PartyIdentification><cac:PartyName><cbc:Name languageID="ENG">Buyer Organization</cbc:Name></cac:PartyName><cac:PostalAddress><cbc:CityName>ROMA</cbc:CityName><cbc:CountrySubentityCode listName="nuts">AT122</cbc:CountrySubentityCode><cac:Country><cbc:IdentificationCode listName="country">BEL</cbc:IdentificationCode></cac:Country></cac:PostalAddress><cac:PartyLegalEntity><cbc:CompanyID>1</cbc:CompanyID></cac:PartyLegalEntity><cac:Contact><cbc:Telephone>05343434</cbc:Telephone><cbc:ElectronicMail>sdsd@sdds.sdsd</cbc:ElectronicMail></cac:Contact></efac:Company><efac:TouchPoint><cac:PartyIdentification><cbc:ID schemeName="touchpoint">TPO-0001</cbc:ID></cac:PartyIdentification></efac:TouchPoint></efac:Organization><efac:Organization><efac:UltimateBeneficialOwner><cbc:ID>UBO-0001</cbc:ID></efac:UltimateBeneficialOwner><efac:Company><cac:PartyIdentification><cbc:ID schemeName="organization">ORG-0002</cbc:ID></cac:PartyIdentification><cac:PartyName><cbc:Name languageID="ENG">Beneficial Owner Organization</cbc:Name></cac:PartyName><cac:PostalAddress><cbc:CityName>VITERBO</cbc:CityName><cbc:CountrySubentityCode listName="nuts">AL033</cbc:CountrySubentityCode><cac:Country><cbc:IdentificationCode listName="country">AUT</cbc:IdentificationCode></cac:Country></cac:PostalAddress><cac:PartyLegalEntity><cbc:CompanyID>2</cbc:CompanyID></cac:PartyLegalEntity><cac:Contact><cbc:Telephone>0113434</cbc:Telephone><cbc:ElectronicMail>vpi@dfdf.dfdf</cbc:ElectronicMail></cac:Contact></efac:Company></efac:Organization><efac:UltimateBeneficialOwner><cbc:ID schemeName="ubo">UBO-0001</cbc:ID></efac:UltimateBeneficialOwner></efac:Organizations></efext:EformsExtension></ext:ExtensionContent></ext:UBLExtension></ext:UBLExtensions><cbc:UBLVersionID>2.3</cbc:UBLVersionID><cbc:CustomizationID>eforms-sdk-1.6</cbc:CustomizationID><cbc:ID schemeName="notice-id">955bf7d9-843a-4194-b7e8-4835e2161385</cbc:ID><cbc:ContractFolderID>c2a0dcd1-4d17-4072-9252-38de16ead921</cbc:ContractFolderID><cbc:IssueDate>2023-03-03+01:00</cbc:IssueDate><cbc:IssueTime>01:06:46+02:00</cbc:IssueTime><cbc:VersionID>01</cbc:VersionID><cbc:RegulatoryDomain>31985R2137</cbc:RegulatoryDomain><cbc:NoticeTypeCode listName="result">can-standard</cbc:NoticeTypeCode><cbc:NoticeLanguageCode listName="language">ENG</cbc:NoticeLanguageCode><cac:ContractingParty><cbc:BuyerProfileURI>http://buyer.df</cbc:BuyerProfileURI><cac:ContractingPartyType><cbc:PartyTypeCode listName="buyer-legal-type">body-pl-ra</cbc:PartyTypeCode></cac:ContractingPartyType><cac:ContractingActivity><cbc:ActivityTypeCode listName="authority-activity">hc-am</cbc:ActivityTypeCode></cac:ContractingActivity><cac:Party><cac:PartyIdentification><cbc:ID schemeName="organization">ORG-0001</cbc:ID></cac:PartyIdentification></cac:Party></cac:ContractingParty><cac:TenderingProcess><cbc:ProcedureCode listName="procurement-procedure-type">comp-tend</cbc:ProcedureCode><cbc:TerminatedIndicator>true</cbc:TerminatedIndicator></cac:TenderingProcess><cac:ProcurementProject><cbc:Name languageID="ENG">PROC TITLE</cbc:Name><cbc:Description languageID="ENG">PROC_DESC</cbc:Description><cbc:ProcurementTypeCode listName="contract-nature">supplies</cbc:ProcurementTypeCode><cac:RequestedTenderTotal><cbc:EstimatedOverallContractAmount currencyID="EUR">100</cbc:EstimatedOverallContractAmount></cac:RequestedTenderTotal><cac:MainCommodityClassification><cbc:ItemClassificationCode listName="cpv">03113000</cbc:ItemClassificationCode></cac:MainCommodityClassification><cac:AdditionalCommodityClassification><cbc:ItemClassificationCode listName="cpv">03110000</cbc:ItemClassificationCode></cac:AdditionalCommodityClassification><cac:RealizedLocation><cac:Address><cbc:CityName>PISA</cbc:CityName></cac:Address></cac:RealizedLocation></cac:ProcurementProject><cac:ProcurementProjectLot><cbc:ID schemeName="Lot">LOT-0001</cbc:ID><cac:TenderingTerms><cbc:FundingProgramCode listName="eu-funded">eu-funds</cbc:FundingProgramCode><cac:AwardingTerms><cac:AwardingCriterion><cac:SubordinateAwardingCriterion><cbc:AwardingCriterionTypeCode listName="award-criterion-type">cost</cbc:AwardingCriterionTypeCode></cac:SubordinateAwardingCriterion></cac:AwardingCriterion></cac:AwardingTerms><cac:AppealTerms><cac:AppealReceiverParty><cac:PartyIdentification><cbc:ID schemeName="touchpoint">TPO-0001</cbc:ID></cac:PartyIdentification></cac:AppealReceiverParty></cac:AppealTerms></cac:TenderingTerms><cac:TenderingProcess><cbc:GovernmentAgreementConstraintIndicator>true</cbc:GovernmentAgreementConstraintIndicator><cac:AuctionTerms><cbc:AuctionConstraintIndicator>true</cbc:AuctionConstraintIndicator></cac:AuctionTerms><cac:ContractingSystem><cbc:ContractingSystemTypeCode listName="framework-agreement">fa-mix</cbc:ContractingSystemTypeCode></cac:ContractingSystem><cac:ContractingSystem><cbc:ContractingSystemTypeCode listName="dps-usage">none</cbc:ContractingSystemTypeCode></cac:ContractingSystem></cac:TenderingProcess><cac:ProcurementProject><cbc:Name languageID="ENG">LOT TITLE</cbc:Name><cbc:Description languageID="ENG">LOT DESCR</cbc:Description><cbc:ProcurementTypeCode listName="contract-nature">supplies</cbc:ProcurementTypeCode><cac:RequestedTenderTotal><cbc:EstimatedOverallContractAmount currencyID="EUR">250</cbc:EstimatedOverallContractAmount></cac:RequestedTenderTotal><cac:MainCommodityClassification><cbc:ItemClassificationCode listName="cpv">03113000</cbc:ItemClassificationCode></cac:MainCommodityClassification><cac:AdditionalCommodityClassification><cbc:ItemClassificationCode listName="cpv">03110000</cbc:ItemClassificationCode></cac:AdditionalCommodityClassification></cac:ProcurementProject></cac:ProcurementProjectLot><cac:TenderResult><cbc:AwardDate>2023-05-05+02:00</cbc:AwardDate></cac:TenderResult></ContractAwardNotice>`

**Comment by elerik** [comment]
Hi. 
Today i just found out (but im not sure, its little bit chaotic) CVS use encoding UTF-16 and Submit use encoding UTF-8 
This mix in encoding made me little bit crazy because I had exactly same issues. 500 again and again. 
So try first Encoding may be its the problem.
Also make sure you send it as "multipart/form-data"

**Comment by rengit62** [comment]
Thanks for your response, but I use the same method for all notices . I just sent notice type 1, 4 and 7, and they where submitted, instead with subtype 29 I get 500 Internal Server Error
Bye
Renato

**Reply by elerik** [reply]
Are you sure that with subtype 29 you are valid ? Because when i create subtype 29 and fill all mandatory fields CVS return me that my xml is not valid against xsd (and as a bonus same result i get via official notice2 web app) 

And it look like problems are in GR-Result group. Where is only one visible mandatory field (BT-142-LotResult) 
no other are required. However xml have to contain also BT-119-LotResult, 
and one of well picture is better than thousands words. All of these fields are not mandatory but validation process ask for them. 

![image](https://github.com/OP-TED/eForms-SDK/assets/11737559/9e08ab51-1511-4738-a416-cd45b68bc0dd)

**Reply by elerik** [reply]
so after i figured out what was vital (`<efac:TenderLot><cbc:ID>LOT-0001</cbc:ID></efac:TenderLot>` and i see you have it in xml) and cvs return its ok. Than subtype 29 was successfully submitted. And i see it also at notice2.preview. Did you managed to fixed it out ?

**Comment by rengit62** [comment, marked as answer]
Yes, thanks !
Renato
